modules/fedimint-mint-client/src/backup.rs

```diff
@@ -41,10 +40,10 @@ impl EcashBackup {
 ///
 /// Used to speed up and improve privacy of ecash recovery,
 /// by avoiding scanning the whole history.
-#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Debug, Encodable, Decodable)]
+#[derive(Clone, PartialEq, Eq, Debug, Encodable, Decodable)]
 pub struct EcashBackupV0 {
     spendable_notes: TieredMulti<SpendableNote>,
-    pending_notes: Vec<(OutPoint, Amount, NoteIssuanceRequest)>,
+    pending_notes: Vec<(OutPoint, NoteIssuanceRequest)>,
```

Comment:
In order to maintain backwards compatibility, each place where we touch an `Encodable` struct, we will unfortunately need to write a db migration to remove the `Amount`. It's definitely needed here, it looks like there are some other structs too (e.g `MintRecoveryState`).

The lightning module has some examples on how to do this, [but they mainly migrate state machines](https://github.com/fedimint/fedimint/blob/master/modules/fedimint-ln-client/src/db.rs). This is just migrating a struct, so hopefully it will be easier. You just need to write a migration that reads the old struct from disk, then re-writes it as the new struct.

Reply:
It's even worse since this is a backup struct and not a SM: we'll have to support the old backup for a long time since we need to retain the ability to read old backups. Why remove that amount here anyway?